translations/translationtool/src/translationtool.php

```diff
@@ -320,31 +320,56 @@ private function createFakeFileForVueFiles() {
 		$fakeFileContent = '';
 
 		foreach ($this->findTranslatableFiles(['.vue']) as $vueFile) {
-			$vueSource = file_get_contents($vueFile);
+			$vueSource = file($vueFile);
```

Comment:
if you read line per line, do multi line `t`/`n` calls still match? I doubt it

try parsing code that says

```js
const text = t(
  'test',
  'my text'
);
```

Reply:
Ahh, you're right yes.

So - the only idea i have then is a quite complex regex, that catches the full block. Just quickly tested this one and it seems to work on my testfile, also with the multiline translations.

Would you be ok with using such a regex? Or do you think it's too complex & error-prone? 🤔 

```
$translatorsRegex = "(\W(\/\/|\/\*|<!--)\s(TRANSLATORS)\s(.+))*"; // Indicating an optional TRANSLATORS-comment
$inbetweenRegex = "(\s*.*\s*)"; // Text between comment and translation-function

preg_match_all("/". $translatorsRegex . $inbetweenRegex . "\Wt\s*\(\s*'([\w]+)',\s*'(.+)'" . "/", $vueSource, $singleQuoteMatches);
```

Reply:
is a regex ever not complex or error-prone? :wink: 

is there any measurable performance impact?

Reply:
So -  implemented and pushed now a fully working version.
However - yes, it seems like this currently has a huge impact on performance... So needs for sure some more love before it is ready to go. 😞